### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Sigma.Basic

### PR DESCRIPTION
**Pull request by imbrem**
---

Currently stuck trying to fill in a few proofs dependent on `tidy`, and getting `mapComp` to compile. Everything else seems to work.

### CODE REVIEW

**Comment by 0art0** [comment]
I noticed that `aesop` (or alternatively `aesop?`) is now able to solve the proofs that used to rely on `tidy` (I recall seeing some Zulip conversations on using `aesop` while porting Category theory, but I can't seem to find any mentioning this specific file - so I thought of mentioning it here).

File: Mathlib/CategoryTheory/Sigma/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+-- Porting note: TODO: don't know how to fix the type errors here
+-- variable {I} {K : Type w₃}
+
+-- /-- The functor `sigma.map` applied to a composition is a composition of functors. -/
+-- @[simps]
+-- def mapComp (f : K → J) (g : J → I) : map (C ∘ g) f ⋙ (map C g : _) ≅ map C (g ∘ f) :=
+--   (descUniq _ _) fun k =>
+--     (isoWhiskerRight (inclCompMap (C ∘ g) f k) (map C g : _) : _) ≪≫ inclCompMap _ _ _
+
+-- #align category_theory.sigma.map_comp CategoryTheory.Sigma.mapComp
```
Comment (Ruben-VandeVelde) [inline Mathlib/CategoryTheory/Sigma/Basic.lean:248]: This one will need fixing, though I'm not quite sure how either.

**Review by Ruben-VandeVelde — CHANGES_REQUESTED** [review]

**Comment by imbrem** [comment]
> I noticed that `aesop` (or alternatively `aesop?`) is now able to solve the proofs that used to rely on `tidy` (I recall seeing some Zulip conversations on using `aesop` while porting Category theory, but I can't seem to find any mentioning this specific file - so I thought of mentioning it here).

I tried using `aesop_cat` but it didn't seem to work

File: Mathlib/CategoryTheory/Sigma/Basic.lean (hunk 1 above)
Comment (nomeata) [reply Mathlib/CategoryTheory/Sigma/Basic.lean]: I think I found the issue.

File: Mathlib/CategoryTheory/Sigma/Basic.lean [hunk 2]
```diff
@@ -31,86 +31,74 @@ variable {I : Type w₁} {C : I → Type u₁} [∀ i, Category.{v₁} (C i)]
...
-lean 3 declaration is
-  forall {I : Type.{u1}} {C : I -> Type.{u3}} [_inst_1 : forall (i : I), CategoryTheory.Category.{u2, u3} (C i)] (X : Sigma.{u1, u3} I (fun (i : I) => C i)), CategoryTheory.Sigma.SigmaHom.{u1, u2, u3} I (fun (i : I) => C i) (fun (i : I) => _inst_1 i) X X
-but is expected to have type
-  forall {I : Type.{u3}} {C : I -> Type.{u1}} [_inst_1 : forall (i : I), CategoryTheory.Category.{u2, u1} (C i)] (X : Sigma.{u3, u1} I (fun (i : I) => C i)), CategoryTheory.Sigma.SigmaHom.{u3, u2, u1} I (fun (i : I) => C i) (fun (i : I) => _inst_1 i) X X
-Case conversion may be inaccurate. Consider using '#align category_theory.sigma.sigma_hom.id CategoryTheory.Sigma.SigmaHom.idₓ'. -/
 /-- The identity morphism on an object. -/
 def id : ∀ X : Σi, C i, SigmaHom X X
-  | ⟨i, X⟩ => mk (𝟙 _)
+  | ⟨_, _⟩ => mk (𝟙 _)
 #align category_theory.sigma.sigma_hom.id CategoryTheory.Sigma.SigmaHom.id
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Sigma/Basic.lean:42]:
```suggestion
#align category_theory.sigma.sigma_hom.id CategoryTheory.Sigma.SigmaHom.idₓ -- Porting note: reordered universes
```

File: Mathlib/CategoryTheory/Sigma/Basic.lean [hunk 3]
```diff
@@ -31,86 +31,74 @@ variable {I : Type w₁} {C : I → Type u₁} [∀ i, Category.{v₁} (C i)]
...
-but is expected to have type
-  forall {I : Type.{u3}} {C : I -> Type.{u1}} [_inst_1 : forall (i : I), CategoryTheory.Category.{u2, u1} (C i)] {X : Sigma.{u3, u1} I (fun (i : I) => C i)} {Y : Sigma.{u3, u1} I (fun (i : I) => C i)} {Z : Sigma.{u3, u1} I (fun (i : I) => C i)}, (CategoryTheory.Sigma.SigmaHom.{u3, u2, u1} I (fun (i : I) => C i) (fun (i : I) => _inst_1 i) X Y) -> (CategoryTheory.Sigma.SigmaHom.{u3, u2, u1} I (fun (i : I) => C i) (fun (i : I) => _inst_1 i) Y Z) -> (CategoryTheory.Sigma.SigmaHom.{u3, u2, u1} I (fun (i : I) => C i) (fun (i : I) => _inst_1 i) X Z)
-Case conversion may be inaccurate. Consider using '#align category_theory.sigma.sigma_hom.comp CategoryTheory.Sigma.SigmaHom.compₓ'. -/
 /-- Composition of sigma homomorphisms. -/
 def comp : ∀ {X Y Z : Σi, C i}, SigmaHom X Y → SigmaHom Y Z → SigmaHom X Z
   | _, _, _, mk f, mk g => mk (f ≫ g)
 #align category_theory.sigma.sigma_hom.comp CategoryTheory.Sigma.SigmaHom.comp
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Sigma/Basic.lean:50]:
```suggestion
#align category_theory.sigma.sigma_hom.comp CategoryTheory.Sigma.SigmaHom.compₓ -- Porting note: reordered universes
```

File: Mathlib/CategoryTheory/Sigma/Basic.lean [hunk 4]
```diff
@@ -31,86 +31,74 @@ variable {I : Type w₁} {C : I → Type u₁} [∀ i, Category.{v₁} (C i)]
...
-theorem assoc : ∀ (X Y Z W : Σi, C i) (f : X ⟶ Y) (g : Y ⟶ Z) (h : Z ⟶ W), (f ≫ g) ≫ h = f ≫ g ≫ h
-  | _, _, _, _, mk f, mk g, mk h => congr_arg mk (Category.assoc _ _ _)
+theorem assoc
+  : ∀ {X Y Z W : Σi, C i} (f : X ⟶ Y) (g : Y ⟶ Z) (h : Z ⟶ W), (f ≫ g) ≫ h = f ≫ g ≫ h
+  | _, _, _, _, mk _, mk _, mk _ => congr_arg mk (Category.assoc _ _ _)
 #align category_theory.sigma.sigma_hom.assoc CategoryTheory.Sigma.SigmaHom.assoc
 
-theorem id_comp : ∀ (X Y : Σi, C i) (f : X ⟶ Y), 𝟙 X ≫ f = f
-  | _, _, mk f => congr_arg mk (Category.id_comp _)
+theorem id_comp : ∀ {X Y : Σi, C i} (f : X ⟶ Y), 𝟙 X ≫ f = f
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Sigma/Basic.lean:67]: Please do not change the binding type of these lemmas.

File: Mathlib/CategoryTheory/Sigma/Basic.lean [hunk 5]
```diff
@@ -31,86 +31,74 @@ variable {I : Type w₁} {C : I → Type u₁} [∀ i, Category.{v₁} (C i)]
...
+  | _, _, mk _ => congr_arg mk (Category.id_comp _)
 #align category_theory.sigma.sigma_hom.id_comp CategoryTheory.Sigma.SigmaHom.id_comp
 
-theorem comp_id : ∀ (X Y : Σi, C i) (f : X ⟶ Y), f ≫ 𝟙 Y = f
-  | _, _, mk f => congr_arg mk (Category.comp_id _)
+theorem comp_id : ∀ {X Y : Σi, C i} (f : X ⟶ Y), f ≫ 𝟙 Y = f
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Sigma/Basic.lean:71]: Idem

File: Mathlib/CategoryTheory/Sigma/Basic.lean [hunk 6]
```diff
@@ -120,29 +108,22 @@ variable {D : Type u₂} [Category.{v₂} D] (F : ∀ i, C i ⥤ D)
...
-Case conversion may be inaccurate. Consider using '#align category_theory.sigma.desc_map CategoryTheory.Sigma.descMapₓ'. -/
+#align category_theory.sigma.nat_trans_app CategoryTheory.Sigma.natTrans_app
+
 /-- (Implementation). An auxiliary definition to build the functor `desc`. -/
 def descMap : ∀ X Y : Σi, C i, (X ⟶ Y) → ((F X.1).obj X.2 ⟶ (F Y.1).obj Y.2)
-  | _, _, sigma_hom.mk g => (F _).map g
+  | _, _, SigmaHom.mk g => (F _).map g
 #align category_theory.sigma.desc_map CategoryTheory.Sigma.descMap
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Sigma/Basic.lean:127]:
```suggestion
#align category_theory.sigma.desc_map CategoryTheory.Sigma.descMapₓ -- Porting note: reordered universes
```

File: Mathlib/CategoryTheory/Sigma/Basic.lean (hunk 4 above)
Comment (nomeata) [reply Mathlib/CategoryTheory/Sigma/Basic.lean:67]: (This was done before I looked at the PR)
With binding type, you mean `lemma` vs. `theorem`?

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**